lib/connections/structs.go

```diff
@@ -40,7 +41,10 @@ type genericDialer interface {
 	String() string
 }
 
-type listenerFactory func(*url.URL, *tls.Config, chan IntermediateConnection, *nat.Service) genericListener
+type listenerFactory interface {
+	New(*url.URL, *tls.Config, chan IntermediateConnection, *nat.Service) genericListener
+	Enabled(config.Configuration) bool
+}
```

Comment:
I think we should pass tye co fig wrapper in the constructor and not pass it in Enabled. This will be needed for utp anyway to lookup stun servers.